### ISSUE TITLE
Why is the exported or custom-trained model size always twice that of COCO pre-trained weights?

### ISSUE THREAD

**Issue by AliBYetiser** [ultralytics/ultralytics#16804]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Whenever I custom train a YOLO model or export a COCO pre-trained weights to any format, the weights of the new model are always around twice the size of the pre-trained weights.
For example
```python
from ultralytics import YOLO
model = YOLO('yolov8x.pt') # 130.5mb
model.export(format='torchscript') # 260.8mb and similar sizes for other exports
```
Could you explain why this is? Are pre-trained weights' quantization and optimizations different from a model obtained from a model.train() or export process? Could you share how pre-trained weights are obtained?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AliBYetiser, thank you for your question and interest in Ultralytics 🚀! An Ultralytics engineer will be with you shortly.

In the meantime, here are a few resources that might help:

- Explore our [Docs](https://docs.ultralytics.com) for more insights on exporting models and weight management. You may find the [Export Tutorial](https://docs.ultralytics.com/modes/export/) particularly useful.
  
- If you suspect a 🐛 bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist in debugging.

- For general questions about custom training or optimizations, our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) guide could be helpful.

Join our community spaces to share insights or get real-time help:
- [Discord](https://ultralytics.com/discord) for chat 🎧
- [Discourse](https://community.ultralytics.com) for discussions
- [Subreddit](https://reddit.com/r/ultralytics) for community threads

## Upgrade

Ensure you're using the latest `ultralytics` package with:

```bash
pip install -U ultralytics
```

## Environments

Run YOLO in one of our verified environments, equipped with all dependencies:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **GCP**: Refer to our [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS**: Learn more in our [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, ensuring the reliability of our platforms.

Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
The pretrained weights are stored as float16. So you have to export with `half=True` to maintain the same size. Not all formats support this.

**Comment by Y-T-G** [comment]
ONNX for example supports `half` export and the size is the same.

```python
>>> from ultralytics import YOLO
>>> model = YOLO("yolo11n.pt")                                         >>> model.export(format="onnx", half=True, device=0)
Ultralytics 8.3.8 🚀 Python-3.10.14 torch-2.4.1+cu121 CUDA:0 (Tesla T4, 14916MiB)
YOLO11n summary (fused): 238 layers, 2,616,248 parameters, 0 gradients, 6.5 GFLOPs

PyTorch: starting from 'yolo11n.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 84, 8400) (5.4 MB)
                                                                       ONNX: starting export with onnx 1.16.2 opset 19...
ONNX: slimming with onnxslim 0.1.34...
ONNX: export success ✅ 1.0s, saved as 'yolo11n.onnx' (5.2 MB)

Export complete (1.2s)
Results saved to /ultralytics
Predict:         yolo predict task=detect model=yolo11n.onnx imgsz=640 half
Validate:        yolo val task=detect model=yolo11n.onnx imgsz=640 data=/usr/src/ultralytics/ultralytics/cfg/datasets/coco.yaml half
Visualize:       https://netron.app
'yolo11n.onnx'
>>> !ls -sh yolo11n.onnx
5.2M yolo11n.onnx
```
